src/site/content/en/blog/building-a-settings-component/index.md

```diff
@@ -0,0 +1,959 @@
+  
+  /* if motion is OK, transition the box-shadow change */
+  @media (--motionOK) {
+    & {
+      transition: box-shadow .1s ease;
+    }
+  }
+
+  /* on hover/active state of parent, increase size prop */
+  @nest input[type="range"]:is(:hover,:active) & {
+    --thumb-highlight-size: 10px;
+  }
+}
+```
+
+{% Aside 'key-term' %} 
+[@custom-media](https://drafts.csswg.org/mediaqueries-5/#custom-mq) 
+is a Level 5 spec addition that
+[PostCSS Custom Media](https://github.com/postcss/postcss-custom-media), part of
+[PostCSS Preset Env](https://preset-env.cssdb.org/features). 
+{% endAside %}
+
+The goal was an easy to manage and animated visual highlight for user feedback.
+By using a box shadow I can prevent effecting layout with the effect. I do this
+by creating a shadow that's not blurred and matches the circular shape of the
+thumb element. Then I change and transition it's spread size on hover.
+
+{% Video 
+  src="video/vS06HQ1YTsbMKSFTIPl2iogUQP73/s835RbH88L5bxjl5bMFl.mp4",
+  className="w-screenshot", 
+  autoplay="true", 
+  loop="true", 
+  muted="true" 
+%}
+
+If only the highlight effect was so easy on checkboxes…
+
+### Cross browser selectors
+
+I found I needed these `-webkit-` and `-moz-` selectors to achieve cross browser
+consistency:
+
+```css
+input[type="range"] {
+  &::-webkit-slider-runnable-track {}
+  &::-moz-range-track {}
+  &::-webkit-slider-thumb {}
+  &::-moz-range-thumb {}
+}
+```
+
+{% Aside 'gotchas' %} 
+[Josh Comeau](https://twitter.com/JoshWComeau) outlines
+why the above examples don't simply use a comma between selectors for cross
+browser styling, see the [Twitter
+thread](https://twitter.com/JoshWComeau/status/1359213591602335752?s=20) for
+more information. 
+{% endAside %}
+
+## Custom Checkbox
+
+Given the following HTML input element, I'll show you how I customized its
+appearance:
+
+```html
+<input type="checkbox">
+```
+
+There's 3 parts to this element we need to customize:
+1. [Checkbox element](#checkbox-element)
+1. [Associated labels](#checkbox-labels)
+1. [Highlight effect](#checkbox-highlight)
+
+### Checkbox element
+
+```css
+input[type="checkbox"] {
+  inline-size: var(--space-sm);   /* increase width */
+  block-size: var(--space-sm);    /* increase height */
+  outline-offset: 5px;            /* focus style enhancement */
+  accent-color: var(--brand);     /* tint the input */
+  position: relative;             /* prepare for an absolute pseudo element */
+  transform-style: preserve-3d;   /* create a 3d z-space stacking context */
+  margin: 0;
+  cursor: pointer;
+}
+```
+
+Besides the pseudo element `transform-style` and `position` styles, it's mostly
+minor opinionated style stuff from me. I like the cursor to be pointer, I like
+outline offsets, default checkboxes are too tiny, and if `accent-color` is
+[supported](https://drafts.csswg.org/css-ui-4/#widget-accent), bring these
+checkboxes into the brand color scheme.
```

Comment:
```suggestion
The `transform-style` and `position` styles prepare for the pseudo-element we will introduce later 
to style the highlight. Otherwise, it's mostly
minor opinionated style stuff from me. I like the cursor to be pointer, I like
outline offsets, default checkboxes are too tiny, and if `accent-color` is
[supported](https://drafts.csswg.org/css-ui-4/#widget-accent), bring these
checkboxes into the brand color scheme.
```

Is this edit correct? I couldn't figure out what you meant about the pseudo-element, but I think this is the ::before later on.